File: Greedy Algorithms/selection_sort.py
def selection_sort(nums):
    n = len(nums)
    
    for i in range(n):
        mini = nums[i]
        index = i

        for j in range(i+1, n):
            if nums[j] < mini:
                mini = nums[j]
                index = j

        temp = nums[i]
        nums[i] = nums[index]
        nums[index] = temp


arr = [1, 0, 99, 5, 2, -99]
selection_sort(arr)
print(arr)

File: Greedy Algorithms/Selection Sort/selection_sort.cpp
#include<iostream>
#include<vector>
using namespace std;

void selectionSort(vector<int> &nums){
    int n = nums.size();

    for(int i=0;i<n;i++){
        int mini = nums[i];
        int index = i;

        for(int j=i+1;j<n;j++){
            if(nums[j] < mini){
                mini = nums[j];
                index = j;
            }
        }

        int temp = nums[i];
        nums[i] = nums[index];
        nums[index] = temp;
    }

    for(int i=0;i<n;i++){
        cout<<nums[i]<<" ";
    }
    cout<<endl;
}


int main(){
    vector<int> v = {1, 0, -99, 99, 5};
    selectionSort(v);

    return 0;
}